### DISCUSSION TITLE
Logging new variables

### DISCUSSION THREAD

**Discussion by SriramKodey** [bitcraze/discussions#1598] [Q&A]
Hi all, I am adding some custom code to the CF firmware. Using the existing logging framework, I want to log some variables from this code. It should be similar to the following segment of code from controller_pid.c. Do you know how I can do this?

```
/**
 * Logging variables for the command and reference signals for the
 * altitude PID controller
 */
LOG_GROUP_START(controller)
/**
 * @brief Thrust command
 */
LOG_ADD(LOG_FLOAT, cmd_thrust, &cmd_thrust)
/**
 * @brief Roll command
 */
/**
 * @brief Desired yaw rate setpoint
 */
LOG_ADD(LOG_FLOAT, yawRate,   &rateDesired.yaw)
LOG_GROUP_STOP(controller)
```

**Comment by gemenerik** [comment]
You can follow the same format as the code you provided. Have you seen the [logging and parameter documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/logparam/)?
